octoprint_authldap/__init__.py

```diff
@@ -0,0 +1,271 @@
+# coding=utf-8
+from __future__ import absolute_import
+
+import octoprint.plugin
+from octoprint.users import UserManager, FilebasedUserManager, User
+from octoprint.settings import settings
+import ldap
+import random
+import string
+import os
+
+
+class LDAPUserManager(FilebasedUserManager, octoprint.plugin.SettingsPlugin, octoprint.plugin.TemplatePlugin):
+
+	# Login phase :
+	# - findUser called, if it return a user
+	# - checkPassword called, if it return True
+	# - login_user called with User returned by previous findUser
+
+	_localUserManager = FilebasedUserManager()
+
+	def __init__(self):
```

Comment:
This function should check if the plugin is working as expected while loading.
If not it should fall back to the FilebasedUserManager.
This could be achieved by raising an exception, which however disables also the settings page, so fixing the configuration from the UI would not be possible.